walk/filesystem.go

```diff
@@ -80,6 +80,11 @@ func (f *FilesystemReader) process() error {
 // Read populates the provided files array with as many files are available until the provided context is cancelled.
 // You must ensure to pass a context with a timeout otherwise this will block until files is full.
 func (f *FilesystemReader) Read(ctx context.Context, files []*File) (n int, err error) {
+	// ensure we record how many files we traversed
+	defer func() {
+		f.stats.Add(stats.Traversed, int32(n))
```

Comment:
Is this right? I see a loop below with cancelation logic. Is it possible that we report the wrong number if we're canceled?

Reply:
Wow, that logic is a bit broken and I don't know how the tests passed.